book/src/design/circuit/gadgets/ecc/addition.md

```diff
@@ -0,0 +1,337 @@
+## Incomplete addition
+- Inputs: $P = (x_p, y_p), Q = (x_q, y_q)$
+- Output: $R = P \;⸭\; Q = (x_r, y_r)$
+
+Formulae:
+- $\lambda \cdot (x_p - x_q) = y_p - y_q$
+- $x_r = \lambda^2 - x_q - x_p$
+- $y_r = \lambda(x_q - x_r) - y_q$
+
+Substituting for $\lambda$, we get the constraints:
+- $(x_r + x_q + x_p) \cdot (x_p - x_q)^2 - (y_p - y_q)^2 = 0$
+  - Note that this constraint is unsatisfiable for $P \;⸭\; (-P)$ (when $P \neq \mathcal{O}$),
+    and so cannot be used with arbitrary inputs.
+- $(y_r + y_q)(x_p - x_q) - (y_p - y_q)(x_q - x_r) = 0$
+
+
+## Complete addition
+
+$\begin{array}{rcll}
+\mathcal{O} &+& \mathcal{O} &= \mathcal{O} ✓\\
+\mathcal{O} &+& (x_q, y_q)  &= (x_q, y_q) ✓\\
+ (x_p, y_p) &+& \mathcal{O} &= (x_p, y_p) ✓\\
+   (x, y)   &+& (x, y)      &= [2] (x, y) ✓\\
+   (x, y)   &+& (x, -y)     &= \mathcal{O} ✓\\
+ (x_p, y_p) &+& (x_q, y_q)  &= (x_p, y_p) \;⸭\; (x_q, y_q), \text{if } x_p \neq x_q ✓
```

Comment:
The ✓ marks were just added for me to keep track of which cases had been checked; they're superfluous now.
```suggestion
\mathcal{O} &+& \mathcal{O} &= \mathcal{O} \\
\mathcal{O} &+& (x_q, y_q)  &= (x_q, y_q) \\
 (x_p, y_p) &+& \mathcal{O} &= (x_p, y_p) \\
   (x, y)   &+& (x, y)      &= [2] (x, y) \\
   (x, y)   &+& (x, -y)     &= \mathcal{O} \\
 (x_p, y_p) &+& (x_q, y_q)  &= (x_p, y_p) \;⸭\; (x_q, y_q), \text{if } x_p \neq x_q
```